pkg/server/server.go

```diff
@@ -105,9 +115,24 @@ func (s *server) RunUntil(stopCh <-chan struct{}) error {
 		return nil
 	}
 
-	// Start serving API and scrape loop
-	go s.runScrape(ctx)
-	return s.GenericAPIServer.PrepareRun().Run(stopCh)
+	g := run.Group{}
+	s.addRunScrape(ctx, &g)
+	s.addGenericAPIServer(stopCh, cancel, &g)
+	if s.metricsAddress != "" {
+		s.addMetricsServer(ctx, &g)
+	}
+
+	fmt.Println("Running group")
```

Comment:
Please use proper logging library

Reply:
Ah, I don't even thing this is necessary, I'll just remove it.